fuel-vm/src/interpreter.rs

```diff
@@ -644,22 +636,25 @@ impl CheckedMetadata for UploadCheckedMetadata {
     }
 }
 
-pub(crate) struct InputContracts<'vm, I> {
-    tx_input_contracts: I,
+pub(crate) struct InputContracts<'vm> {
+    input_contracts: &'vm alloc::collections::BTreeSet<ContractId>,
     panic_context: &'vm mut PanicContext,
 }
 
-impl<'vm, I: Iterator<Item = &'vm ContractId>> InputContracts<'vm, I> {
-    pub fn new(tx_input_contracts: I, panic_context: &'vm mut PanicContext) -> Self {
+impl<'vm> InputContracts<'vm> {
+    pub fn new(
+        input_contracts: &'vm alloc::collections::BTreeSet<ContractId>,
+        panic_context: &'vm mut PanicContext,
+    ) -> Self {
         Self {
-            tx_input_contracts,
+            input_contracts,
             panic_context,
         }
     }
 
     /// Checks that the contract is declared in the transaction inputs.
     pub fn check(&mut self, contract: &ContractId) -> SimpleResult<()> {
-        if !self.tx_input_contracts.any(|input| input == contract) {
+        if !self.input_contracts.contains(contract) {
```

Comment:
Essentially, the idea of this PR is making this lookup more performant? Is that right?

Reply:
Yep